src/BenchmarksApps/SslStream/Client/Program.cs

```diff
@@ -0,0 +1,249 @@
+                s_counters.BytesWritten += bufferSize;
+            }
+
+            sw.Stop();
+            LogMetric("sslstream/write/mean", s_counters.BytesWritten / sw.Elapsed.TotalSeconds);
+        }
+
+        static async Task ReadingTask(SslStream stream, int bufferSize, CancellationToken cancellationToken = default)
+        {
+            if (bufferSize == 0)
+            {
+                return;
+            }
+
+            BenchmarksEventSource.Register("sslstream/read/mean", Operations.Avg, Operations.Avg, "Mean bytes read per second.", "Bytes per second - mean", "n2");
+
+            var recvBuffer = new byte[bufferSize];
+            Stopwatch sw = Stopwatch.StartNew();
+            bool isWarmup = true;
+
+            while (s_isRunning)
+            {
+                if (isWarmup && !s_isWarmup)
+                {
+                    isWarmup = false;
+                    sw.Restart();
+                    s_counters.BytesRead = 0;
+                }
+
+                int bytesRead = await stream.ReadAsync(recvBuffer, cancellationToken).ConfigureAwait(false);
+                s_counters.BytesRead += bytesRead;
+            }
+
+            sw.Stop();
+            LogMetric("sslstream/read/mean", s_counters.BytesRead / sw.Elapsed.TotalSeconds);
+        }
+
+        using var sock = new Socket(SocketType.Stream, ProtocolType.Tcp);
+        await sock.ConnectAsync(options.Hostname, options.Port).ConfigureAwait(false);
+        using var stream = await EstablishSslStreamAsync(sock, options).ConfigureAwait(false);
+
+        byte[] recvBuffer = new byte[options.ReceiveBufferSize];
+
+        s_isRunning = true;
+        var writeTask = Task.Run(() => WritingTask(stream, options.SendBufferSize));
+        var readTask = Task.Run(() => ReadingTask(stream, options.ReceiveBufferSize));
+
+        await Task.Delay(options.Warmup).ConfigureAwait(false);
+        Log("Completing warmup...");
+
+        s_isWarmup = false;
+        await Task.Delay(options.Duration).ConfigureAwait(false);
+        Log("Completing scenario...");
+        s_isRunning = false;
+        await Task.WhenAll(writeTask, readTask).ConfigureAwait(false);
+    }
+
+    static SslClientAuthenticationOptions CreateSslClientAuthenticationOptions(ClientOptions options)
+    {
+        var sslOptions = new SslClientAuthenticationOptions
+        {
+            TargetHost = options.TlsHostName ?? options.Hostname,
+            RemoteCertificateValidationCallback = delegate { return true; },
+            ApplicationProtocols = new List<SslApplicationProtocol> { new(options.Scenario.ToString()) },
+#if NET8_0_OR_GREATER
+            AllowTlsResume = options.AllowTlsResume,
+#endif
+            EnabledSslProtocols = options.EnabledSslProtocols,
+            CertificateRevocationCheckMode = options.CertificateRevocationCheckMode,
+        };
+
+        if (options.ClientCertificate != null)
+        {
+            switch (options.CertificateSelection)
```

Comment:
I'm not sure if we need all this but I'm fine if we keep it. I would expect the differences would be minimal. I would expect the biggest difference would be certificate or not.

Reply:
I would also expect differences in Context vs non-context. And once we implement TLS resume for client certs (https://github.com/dotnet/runtime/issues/94561), then I would expect cert callback to be the slowest.